### PR TITLE
LPS-163767 Make sure to return a valid defaultLanguaId

### PR DESCRIPTION
**Pull request by ces-huynguyen**
LPS-163767 Make sure to return a valid defaultLanguaId

### CODE REVIEW

**Comment by ces-huynguyen** [comment]
ci:test

**Comment by rodrigopaulino** [comment]
ci:stop

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContext.java [hunk 1]
```diff
@@ -565,7 +565,11 @@ public String getDefaultLanguageId() {
 		String defaultLanguageId = _getFormBuilderContextDefaultLanguageId();
 
 		if (defaultLanguageId != null) {
-			return defaultLanguageId;
+			Locale defaultLocale = LocaleUtil.fromLanguageId(defaultLanguageId);
+
+			if (ArrayUtil.contains(getAvailableLocales(), defaultLocale)) {
```
[the remark this answers is not in the document]
Comment (tototrinh) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContext.java:570]: Hi @rodrigopaulino, thanks for your question.
> Why did you choose to include this validation?

Avoid _sortedLocales_ contains an undefined value
https://github.com/liferay/liferay-portal-ee/blob/master/modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js#L45

> What happens if a form's default language is removed from the available language?
> Does the form change its default language?

We got the same issue. _availableLocalesArray_ is [] and _sortedLocales_ will contain undefine value and errors will be thrown. The form couldn't change its default language due to errors.
I think the change from [LPS-133578](https://issues.liferay.com/browse/LPS-133578) (https://github.com/liferay/liferay-portal-ee/blob/master/modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js#L41-L50) cause the issue.
Sorry, I didn't test this case. I only focused on defaultLanguageId.

**Comment by tototrinh** [comment]
Hi @rodrigopaulino,
I update the code, please help take a look. Thanks
with this change:
> What happens if a form's default language is removed from the available language?

before: jp_JP is a defaultLanguageId
image

after: the form is shown successfully without errors.
image

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hey, @tototrinh. Thanks for taking care of the other issue.

In regards to your proposed solution, this changes a bit of the logic here. The idea is to verify if the context's default language is not valid, then check if the form's default language is not valid, then send the site's default. With this change, you're just verifying if the context's default language is not valid then returning the site's default language.
Therefore, my suggestion is to simply add in each if statement of this method the following call: LanguageUtil.isAvailableLocale(String).

Hope this helps.

**Comment by tototrinh** [comment]
Hi @rodrigopaulino, Thank you for the clarification, it makes sense. I have updated the code. Please help review again. Thanks.

**Review by rodrigopaulino — APPROVED** [review]
Hi, @tototrinh. LGTM from the Forms perspective. 
Did you send these changes in the translation manager to be reviewed by the frontend team first? If not, could you please close this one and send them?
Thx.

Thx!

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by tototrinh** [comment]
Hi @rodrigopaulino, do you mean these changes should be sent to https://github.com/liferay-frontend/liferay-portal?

**Comment by rodrigopaulino** [comment]
That's right, @tototrinh. I'll move on and close this so you can send your PR for them to review as well. Thanks.

**Pull request closed without merging**